sdk/tests/iot/common/CMakeLists.txt

```diff
@@ -7,15 +7,18 @@ project (az_iot_common_test LANGUAGES C)
 
 set(CMAKE_C_STANDARD 99)
 
-include(AddTestCMocka)
+include(AddCMockaTest)
 
 create_map_file(az_iot_common_test.map)
 
 add_cmocka_test(az_iot_common_test SOURCES
                 main.c
                 test_az_iot_common.c
-                COMPILE_OPTIONS ${DEFAULT_C_COMPILE_FLAGS}
-                LINK_TARGETS
+                COMPILE_OPTIONS ${DEFAULT_C_COMPILE_FLAGS} ${NO_CLOBBERED_WARNING}
+                LINK_LIBRARIES ${CMOCKA_LIBRARIES}
                     az_iot_common
                     az_core
+                INCLUDE_DIRECTORIES ${CMOCKA_INCLUDE_DIR}
                 )
+
+add_cmocka_test_environment(az_iot_common_test)
```

Comment:
What's the purpose of doing these function calls?

Reply:
Did we add these for all test cmakelists? Do we need to do the same for samples?

Reply:
The CMake module offered by Cmoka gives us this function as well and they used it on their samples.
This is for working with cmake cross-compiler (using a win-emulator like wine).  
I added to all the calls we do to create a cmoka test (iot and core) so we basically follow the cmocka examples (https://gitlab.com/cmocka/cmocka/-/blob/master/example/CMakeLists.txt#L16)